src/v/security/audit/schemas/application_activity.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "schemas.h"
+
+namespace security::audit {
+class api_activity : public ocsf_base_event {
+public:
+    enum class activity_id : int {
+        unknown = 0,
+        create = 1,
+        read = 2,
+        update = 3,
+        delete_id = 4,
+        other = 99
+    };
+
+    enum class status_id : int {
+        unknown = 0,
+        success = 1,
+        failure = 2,
+        other = 99
+    };
+
+    api_activity(
+      activity_id activity_id,
+      actor&& actor,
+      api&& api,
+      network_endpoint&& dst_endpoint,
+      std::optional<http_request>&& http_request,
+      fragmented_vector<resource_detail> resources,
+      enum severity_id severity_id,
+      network_endpoint&& src_endpoint,
+      status_id status_id,
+      timestamp_t time,
+      api_activity_unmapped&& unmapped)
+      : ocsf_base_event(
+        category_uid::application_activity,
+        class_uid::api_activity,
+        severity_id,
+        time,
+        security::audit::type_uid(security::audit::type_uid::type(activity_id)))
+      , activity_id(activity_id)
+      , actor(std::move(actor))
+      , api(std::move(api))
+      , dst_endpoint(std::move(dst_endpoint))
+      , http_request(std::move(http_request))
+      , resources(std::move(resources))
+      , src_endpoint(std::move(src_endpoint))
+      , status_id(status_id)
+      , unmapped(std::move(unmapped)) {
+        _key = hash();
+    }
+
+    auto equality_fields() const {
+        return std::tuple_cat(
+          std::tie(
+            activity_id,
+            actor,
+            api,
+            dst_endpoint.ip,
+            http_request,
+            resources,
+            src_endpoint.ip,
+            status_id,
+            unmapped),
+          ocsf_base_event::equality_fields());
+    }
+
+    size_t hash() final { return std::hash<api_activity>()(*this); }
+
+private:
+    activity_id activity_id;
+    actor actor;
+    api api;
+    network_endpoint dst_endpoint;
+    std::optional<http_request> http_request;
+    fragmented_vector<resource_detail> resources;
+    network_endpoint src_endpoint;
+    status_id status_id;
+    api_activity_unmapped unmapped;
+
+    friend void rjson_serialize(
+      ::json::Writer<::json::StringBuffer>& w, const api_activity&);
```

Comment:
This is a bit of a nit, but there are compile time benefits to defining this inline; i.e., as a hidden friend.